src/Components/dashboard/route/UpdateRouteButton.jsx

```diff
@@ -0,0 +1,247 @@
+import { Box, ListItemText, Typography } from "@mui/material";
+import { Button } from "@mui/material";
+import Sidebar from "../sidebar/Sidebar.jsx";
+import Popup from 'reactjs-popup';
+import 'reactjs-popup/dist/index.css';
+
+import "../Dashboard.css";
+import { Link } from "react-router-dom";
+
+import AddIcon from "@mui/icons-material/Add";
+import List from "@mui/material/List";
+
+import React, { useState } from "react";
+import TextField from "@mui/material/TextField";
+
+import Buttons from "../operatorsanddriver/Button.jsx";
+import "./UpdateRoute.css";
+import ListItem from "@mui/material/ListItem";
+import { useDispatch, useSelector } from "react-redux";
+import { updateRoute } from "../../../redux/Action/routes.js";
+import { Dialog } from "@material-ui/core";
+
+const un = "Bus stop one, You can add others by hitting + button"
+let id = location.href.split("?id=")[1];
+let routeUuid = decodeURIComponent(id);
+
+const UpdateRouteButton = (props) => {
+
+    const [routeCode, setRouteID] = useState("");
+    const [startLocation, setStartPoint] = useState("");
+    const [endLocation, setEndPoint] = useState("");
+    const [stop, setStop] = useState("");
+
+    const routeDetails = useSelector((state) => state.routeDetails);
+    const {loading, route, error } = routeDetails
+
+    
+   
+   
+
+    const [routeIDError, setRouteIDError] = useState(false);
+    const [startPointError, setStartPointError] = useState(false);
+    const [endPointError, setEndPointError] = useState(false);
+    const [stopError, setStopError] = useState(false);
+
+    const [openUpdater, setOpenUpdater] = useState(true);
+    const [open, setOpen] = useState(false);
+    const [updated, setupdated] = useState(false);
+
+    const handleClear = () => {
+        console.log("Content cleared");
```

Comment:
Most of your functions returns console logs messages in this file, hope to improve it better you may remove these lines too.

Reply:
Thank you for the comment @abraham-kwizera , I have resolved this